src/lib/asyncItemIterator.ts

```diff
@@ -175,44 +176,64 @@ async function process(
 
   const orb1 = ora(
     `[${scanned}/${repos.length}] Scanning repos for ${
-      processIssues ? 'issues' : 'PR'
+      type === 'issues' ? 'issues' : 'PR'
     }s`
   ).start();
 
   // Concurrently find all PRs or issues in all relevant repositories
+  let cached = 0;
   const q = new Q({concurrency});
   q.addAll(
     repos.map(repo => {
       return async () => {
         try {
           let localItems;
-          if (processIssues) {
-            localItems = await retryException<Issue[]>(async () => {
-              if (delay) await delayMs(nextDelay(delay));
-              return await repo.listIssues();
-            }, retryStrategy);
-          } else {
-            localItems = await retryException<PullRequest[]>(async () => {
-              if (delay) await delayMs(nextDelay(delay));
-              return await repo.listPullRequests();
-            }, retryStrategy);
+          if (!cli.flags.nocache) {
+            const cachedData = await readFromCache(repo, type);
+            if (cachedData !== null) {
+              localItems =
+                (type === 'issues' ? cachedData.issues : cachedData.prs) ?? [];
+              ++cached;
+            }
+          }
+          if (!localItems) {
+            if (type === 'issues') {
+              localItems = await retryException<Issue[]>(async () => {
+                if (delay) await delayMs(nextDelay(delay));
+                return await repo.listIssues();
+              }, retryStrategy);
+              await saveToCache(repo, type, {prs: [], issues: localItems});
+            } else {
+              localItems = await retryException<PullRequest[]>(async () => {
+                if (delay) await delayMs(nextDelay(delay));
+                return await repo.listPullRequests();
+              }, retryStrategy);
+              await saveToCache(repo, type, {prs: localItems, issues: []});
+            }
           }
           items.push(
             ...localItems.map(item => {
               return {repo, item};
             })
           );
           scanned++;
-          orb1.text = `[${scanned}/${repos.length}] Scanning repos for PRs`;
+          orb1.text = `[${scanned}/${repos.length}] Scanning repos for ${
+            type === 'issues' ? 'issue' : 'PR'
```

Comment:
not familiar with type annotations in TypeScript and I'm really curious, is it possible to use `String(type)` here instead? would simplify these quite a bit

Reply:
It's a ternary operator

Reply:
I mean it's simple enough imo

Reply:
Yes, could just use `${type}` (`type` is already a string internally), it's just singular "issue" vs plural "issues" being handled here. I didn't invent anything new here, I just copy-pasted this from another place in the code, there are quite a lot of those.